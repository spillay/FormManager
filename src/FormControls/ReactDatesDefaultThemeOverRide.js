import React from 'react'
import ThemedStyleSheet from 'react-with-styles/lib/ThemedStyleSheet';
import aphroditeInterface from 'react-with-styles-interface-aphrodite';
import { css, withStyles } from 'react-with-styles';
import DefaultTheme from 'react-dates/lib/theme/DefaultTheme';

ThemedStyleSheet.registerInterface(aphroditeInterface);
const ReactDatesDefaultThemeOverRide = ThemedStyleSheet.registerTheme({
  reactDates: {
    ...DefaultTheme.reactDates,
    // color: {
    //   ...DefaultTheme.reactDates.color,
    //   // backgroundFocused: '#0187b4',
    //   placeholderText: '#0187b4',
    // },
    sizing: {
      inputWidth: 105,
      arrowWidth: 20,
    },

    font: {
      size: 12,
      captionSize: 12,
      input: {
        size: 13,
        lineHeight: '13px',
        size_small: 13,
        lineHeight_small: '13px',

      },
    },

  },
});

export { css, withStyles, ReactDatesDefaultThemeOverRide };