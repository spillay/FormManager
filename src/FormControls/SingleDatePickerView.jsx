import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { SingleDatePicker } from 'react-dates';
import moment from 'moment';

export class SingleDatePickerView extends React.Component {

    constructor(props) {
        super(props);
        //this.state = {date:moment()}
        this.state = {date:null}
    }
    onDateChange = (date) => {
        // console.log(`onDateChange key :${this.props._key}`," date ", date)
        this.setState({date:date});
        this.props.onChange({target:{value:date.format("MMM Do YYYY")}},this.props._key)
    } 
    onFocusChange = ({ focused }) => {
        this.setState({ focused })
    }

    render() {
        return (
            <div key={this.props._key} >
                <SingleDatePicker
                    date={this.state.date} // momentPropTypes.momentObj or null  current date
                    onDateChange={this.onDateChange} // PropTypes.func.isRequired
                    focused={this.state.focused} // PropTypes.bool
                    onFocusChange={this.onFocusChange}
                    {...this.props.options}                 
                />
                <div className="validation-error" >{this.props.errorFor(this.props._key)}</div>
                <small className="form-text text-muted ">( Please provide valid {this.props.label}. )</small>
            </div>
        )
    }


} 
SingleDatePickerView.propTypes = {
    _key: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    errorFor: PropTypes.func.isRequired
}
