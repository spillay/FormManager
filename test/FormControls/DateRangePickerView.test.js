import React from 'react';
import 'react-dates/initialize';
import {DateRangePickerView} from '../../src/FormControls/DateRangePickerView'
import renderer from 'react-test-renderer';

test('DateRangePickerView test', () => {
  const baseProps = {
    _key: "dateRange",
    label: "Date Range",
    onChange: () => {console.log("onChange")},
    errorFor: () => {console.log("Error")},
  };
  const component = renderer.create(
    <DateRangePickerView {...baseProps}/>,
  );
  let tree = component.toJSON();
  console.log(expect(tree));
  expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   tree.props.onMouseEnter();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   tree.props.onMouseLeave();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
});