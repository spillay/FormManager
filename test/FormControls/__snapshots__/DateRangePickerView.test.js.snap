// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePickerView test 1`] = `
<div>
  <div
    className="DateRangePicker DateRangePicker_1"
  >
    <div>
      <div
        className="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2"
      >
        <div
          className="DateInput DateInput_1"
        >
          <input
            aria-describedby="DateInput__screen-reader-message-startDateId"
            aria-label="Start Date"
            autoComplete="off"
            className="DateInput_input DateInput_input_1"
            disabled={false}
            id="startDateId"
            name="startDateId"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Start Date"
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
          <p
            className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
            id="DateInput__screen-reader-message-startDateId"
          >
            Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
          </p>
        </div>
        <div
          aria-hidden="true"
          className="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
          role="presentation"
        >
          <svg
            className="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
            viewBox="0 0 1000 1000"
          >
            <path
              d="M694.4 242.4l249.1 249.1c11 11 11 21 0 32L694.4 772.7c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210.1-210.1H67.1c-13 0-23-10-23-23s10-23 23-23h805.4L662.4 274.5c-21-21.1 11-53.1 32-32.1z"
            />
          </svg>
        </div>
        <div
          className="DateInput DateInput_1"
        >
          <input
            aria-describedby="DateInput__screen-reader-message-endDateId"
            aria-label="End Date"
            autoComplete="off"
            className="DateInput_input DateInput_input_1"
            disabled={false}
            id="endDateId"
            name="endDateId"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="End Date"
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
          <p
            className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
            id="DateInput__screen-reader-message-endDateId"
          >
            Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="validation-error"
  />
  <small
    className="form-text text-muted "
  >
    ( Please provide a start and end date for 
    Date Range
    . )
  </small>
</div>
`;
